From: Aron Griffis <agriffis@redhat.com>
Date: Thu, 3 Apr 2008 19:26:56 -0400
Subject: [xen] ia64: add srlz instruction to asm
Message-id: 20080403232655.GB15393@redhat.com
O-Subject: [RHEL5.2 PATCH][XEN] add srlz instruction to ia64 asm
Bugzilla: 440261
RH-Acked-by: Jarod Wilson <jwilson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Bill Burns <bburns@redhat.com>

Bugzilla Bug 440261: xen/ia64 asm missing srlz instruction
https://bugzilla.redhat.com/show_bug.cgi?id=440261

Summary
-------
This patch was added to xen-unstable over a year ago to fix
a compile-time warning:

arch/ia64/xen/xenentry.S: Assembler messages:
arch/ia64/xen/xenentry.S:128: Warning: Use of 'st4' violates RAW dependency 'DTC' (data)

This is an error in the assembly that can fault if the tr is
referenced without a prior srlz (or instruction implying srlz).

Upstream
--------
http://xenbits.xensource.com/xen-unstable.hg?rev/0df9dc2f1d03

Testing
-------
As mentioned, this has been upstream for more than a year.
Additionally I built 2.6.18-87.el5 with this patch.  The warning
disappeared and the resulting kernel boots on HP Integrity
rx6600.  I don't have an actual scenario cooked up to demonstrate
the problem, other than the obvious warning...

linux-2.6-xen-ia64-srlz-in-xenentry.patch
-----------------------------------------

diff --git a/arch/ia64/xen/xenentry.S b/arch/ia64/xen/xenentry.S
index 41a9d28..0ce51a4 100644
--- a/arch/ia64/xen/xenentry.S
+++ b/arch/ia64/xen/xenentry.S
@@ -122,6 +122,8 @@ GLOBAL_ENTRY(ia64_switch_to)
 	;;
 	itr.d dtr[r25]=r23		// wire in new mapping...
 #ifdef CONFIG_XEN
+	;;
+	srlz.d
 	mov r9=1
 	movl r8=XSI_PSR_IC
 	;;
